### ISSUE TITLE
Extract after rotate() acts on Pre instead of Post

### ISSUE THREAD

**Issue by turadg** [lovell/sharp#145]
The `rotate()` from #34 is very useful, however I think the implementation may have missed a case: `extract()` gets the same pixels before a `rotate()` as after. Reading the docs, it does only say that pre/post `resize()` would matter, but I think it's fair to expect that someone would want pre/post `rotate()` as well.

Sample code:
```
  var e = sharp(imagePath)
  	.rotate(img.rotation)
//        .resize(img.masterWidth, img.masterHeight)
        .extract(region.top, region.left, region.width, region.height);
  console.log('extract options', e.options);
  e
    .toFile(output);
```

If you run this without the `resize()` you'll see that the extract options have Pre offsets, even though it's chained after the rotation. (Moving `extract()` before the rotation gives the same output.) A work-around is to resize to the full dimensions before the extract.

**Comment by lovell** [comment]
Hi Turadg, you're quite right, a rotation (either via an explicit angle or EXIF-based automagic) currently always occurs after any pre-resize extraction. Good spot; thanks for reporting this.

**Comment by lovell** [comment]
As the fix for this problem could potentially break people's existing use of the API, I've added it to the `intake` branch for inclusion in the next major release, v0.9.0.

Anyone needing this feature sooner can use `npm install lovell/sharp#intake`.

Thanks again @turadg for reporting this.

**Comment by turadg** [comment]
Thanks for the quick resolution!

**Comment by lovell** [comment]
The fix is now in master and will be included in v0.9.0.
